crits/samples/migrate.py

```diff
@@ -1,9 +1,94 @@
+def migrate_backdoors(self):
+    """
+    Create backdoor objects from backdoors on samples.
+    """
+
+    if not self.unsupported_attrs:
+        return
+
+    if 'backdoor' not in self.unsupported_attrs:
+        return
+
+    print "Migrating backdoor for %s" % self.id
+
+    from crits.backdoors.handlers import add_new_backdoor
+    backdoor = self.unsupported_attrs['backdoor']
+    name = backdoor.get('name', '')
+    version = backdoor.get('version', None)
+    if not name:
+        return
+
+    # Create a new backdoor family, and if we have a version the more specific
+    # backdoor will be created too. Use source and campaign of the current
+    # Sample.
+    result = add_new_backdoor(name,
+                              version=version,
+                              source=self.source,
+                              campaign=self.campaign)
+    if result['success']:
+        self.add_relationship(rel_item=result['object'],
+                              rel_type="Related_To",
+                              rel_reason="Migrated")
+        # Save the backdoor after relationship was created.
+        result['object'].save()
+    else:
+        print "Error migrating %s: %s" % (self.id, result['message'])
+
+def migrate_exploits(self):
+    """
+    Create exploit objects from exploits on samples.
+    """
+
+    if not self.unsupported_attrs:
+        return
+
+    if 'exploit' not in self.unsupported_attrs:
+        return
+
+    from crits.exploits.handlers import add_new_exploit
+    exploits = self.unsupported_attrs['exploit']
+    for exp in exploits:
+        print "Migrating exploit for %s" % self.id
+
+        # Create a new exploit object. Use the source and campaign from the
+        # current sample. The "old" exploit format was a list of dictionaries
+        # with the key of "cve" and a value that we will use for name and CVE.
+        result = add_new_exploit(exp['cve'],
+                                 cve=exp['cve'],
+                                 source=self.source,
+                                 campaign=self.campaign)
+        if result['success']:
+            self.add_relationship(rel_item=result['object'],
+                                  rel_type="Related_To",
+                                  rel_reason="Migrated")
+            # Save the exploit after relationship was created.
+            result['object'].save()
+        else:
+            print "Error migrating %s: %s" % (self.id, result['message'])
+
+
+
 def migrate_sample(self):
     """
     Migrate to the latest schema version.
     """
 
-    migrate_2_to_3(self)
+    migrate_3_to_4(self)
+
+def migrate_3_to_4(self):
+    """
+    Migrate from schema 3 to 4.
+    """
+
+    if self.schema_version < 3:
+        migrate_2_to_3(self)
+
+    if self.schema_version == 3:
+        migrate_backdoors(self)
+        migrate_exploits(self)
+        self.schema_version = 4
+        self.save()
+        self.reload()
```

Comment:
I *think* you can remove the `.save()` and `.reload()` from the previous 2_to_3 migration since it's being done here.